src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+interface Product {
+  id: number;
+  title: string;
+  thumbnail: string;
+  description: string;
+  price: number;
+}
+
+function ProductDetails() {
+  const [product, setProduct] = useState<Product | null>(null);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [price, setPrice] = useState(0);
+  const { id } = useParams<{ id: string }>();
+
+  useEffect(() => {
+    fetch(`https://dummyjson.com/products/${id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        setProduct(data);
+        setTitle(data.title);
+        setDescription(data.description);
+        setPrice(data.price);
+      })
+      .catch((error) => console.log(error));
+  }, [id]);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setDescription(event.target.value);
+  };
+
+  const handlePriceChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPrice(parseFloat(event.target.value));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const updatedProduct = { ...product, name: title, description, price };
```

Comment:
the reason the update kept failing is because you are sending the entire product object including the `id` in the body.
you should only send the updated texts (`title`, `description` and `price`).

also, i'm not sure why you changed the key `title` to `name`